modules/useQueries.js

```diff
@@ -26,25 +28,40 @@ function useQueries(createHistory) {
       parseQueryString = defaultParseQueryString
 
     function addQuery(location) {
-      if (location.query == null)
-        location.query = parseQueryString(location.search.substring(1))
+      if (location.query == null) {
+        const { search } = location
+        location.query = parseQueryString(search.substring(1))
+        location[SEARCH_BASE_KEY] = { search, searchBase: '' }
```

Comment:
The point of this logic is exactly to allow that overwriting (not exactly overwriting but close).